File: 1.0-northcote/java/uk/co/bwhazel/jcounterpoise/dataformatters/XmlDataFormatter.java
/**
 * Copyright: (c) Benedict W. Hazel, 2011-2012
 * JsonDataFormatter: Class to export counterpoise correction data to XML
 * format.
 */

package uk.co.bwhazel.jcounterpoise.dataformatters;

import java.io.IOException;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import uk.co.bwhazel.jcounterpoise.IEncounter;

/**
 * Exports counterpoise correction data to extensible markup language format
 * (XML).
 * @author Benedict Hazel
 */
public class XmlDataFormatter implements IDataFormatter {

    /**
     * Exports the counterpoise correction calculation data in extensible markup
     * language format (XML).
     * @param encounter Instance of class implementing IEncounter containing the
     * calculation data.
     * @param stream Stream to write formatted data into.
     * @throws IOException Exception thrown if an error occurs while exporting
     * data to XML format.
     */
    public void exportData(IEncounter encounter, OutputStream stream)
            throws IOException {
        OutputStreamWriter xml = new OutputStreamWriter(stream);
        xml.write("<?xml version=\"1.0\" encoding=\"utf-8\"?>");
        xml.write("<enc:Counterpoise xmlns:enc="
                + "\"http://encounter.codeplex.com\" enc:Description=\"");
        xml.write(encounter.getDescription().concat("\">"));
        xml.write("<enc:Basis enc:Type=\"Dimer\">");
        xml.write("<enc:Dimer>");
        if (encounter.getEnergyCount() >= 1) {
            xml.write(String.valueOf(encounter.getDimer()));
        }
        xml.write("</enc:Dimer>");
        xml.write("<enc:MonomerA>");
        if (encounter.getEnergyCount() >= 2) {
            xml.write(String.valueOf(encounter.getMonomerADimerBasis()));
        }
        xml.write("</enc:MonomerA>");
        xml.write("<enc:MonomerB>");
        if (encounter.getEnergyCount() >= 3) {
            xml.write(String.valueOf(encounter.getMonomerBDimerBasis()));
        }
        xml.write("</enc:MonomerB>");
        xml.write("</enc:Basis>");
        xml.write("<enc:Basis enc:Type=\"Monomer\">");
        xml.write("<enc:MonomerA>");
        if (encounter.getEnergyCount() >= 4) {
            xml.write(String.valueOf(encounter.getMonomerAMonomerBasis()));
        }
        xml.write("</enc:MonomerA>");
        xml.write("<enc:MonomerB>");
        if (encounter.getEnergyCount() == 5) {
            xml.write(String.valueOf(encounter.getMonomerBMonomerBasis()));
        }
        xml.write("</enc:MonomerB>");
        xml.write("</enc:Basis>");
        xml.write("<enc:InteractionEnergy>");
        xml.write("<enc:Hartree>");
        if (encounter.getEnergyCount() >= 3) {
            xml.write(String.valueOf(encounter.getInteractionEnergyHartrees()));
        }
        xml.write("</enc:Hartree>");
        xml.write("<enc:Kjmol>");
        if (encounter.getEnergyCount() >= 3) {
            xml.write(String.valueOf(encounter.getInteractionEnergyKjmol()));
        }
        xml.write("</enc:Kjmol>");
        xml.write("</enc:InteractionEnergy>");
        xml.write("<enc:BindingConstant>");
        if (encounter.getEnergyCount() >= 3) {
            xml.write(String.valueOf(encounter.getBindingConstant()));
        }
        xml.write("</enc:BindingConstant>");
        xml.write("</enc:Counterpoise>");
        xml.close();
    }
}

File: 1.0-northcote/cs/Sharpen/DataFormatters/XmlDataFormatter.cs
// <copyright file="XmlDataFormatter.cs" company="Benedict W. Hazel">
//      Benedict W. Hazel, 2011-2012
// </copyright>
// <author>Benedict W. Hazel</author>
// <summary>
//      XmlDataFormatter: Class to export counterpoise correction data to XML format.
// </summary>

using System.IO;
using System.Xml;

namespace BWHazel.Sharpen.DataFormatters
{
    /// <summary>
    /// Exports counterpoise correction data to XML format.
    /// </summary>
    public class XmlDataFormatter : IDataFormatter
    {
        /// <summary>
        /// Exports the counterpoise correction calculation data in XML format.
        /// </summary>
        /// <param name="encounter">Instance of class implementing <see cref="IEncounter"/> containing the calculation data.</param>
        /// <param name="stream">Stream to write XML formatted data into.</param>
        public void ExportData(IEncounter encounter, Stream stream)
        {
            StreamWriter xml = new StreamWriter(stream);
            string xmlns = @"http://encounter.codeplex.com";
            XmlWriter writer = XmlWriter.Create(xml);
            writer.WriteStartDocument();
            writer.WriteStartElement("enc", "Counterpoise", xmlns);
            writer.WriteStartAttribute("enc", "Description", xmlns);
            writer.WriteValue(encounter.Description);
            writer.WriteEndAttribute();
            writer.WriteStartElement("enc", "Basis", xmlns);
            writer.WriteStartAttribute("enc", "Type", xmlns);
            writer.WriteValue("Dimer");
            writer.WriteEndAttribute();
            writer.WriteStartElement("enc", "Dimer", xmlns);
            if (encounter.EnergyCount >= 1)
            {
                writer.WriteValue(encounter.Dimer.ToString());
            }

            writer.WriteEndElement();
            writer.WriteStartElement("enc", "MonomerA", xmlns);
            if (encounter.EnergyCount >= 2)
            {
                writer.WriteValue(encounter.MonomerADimerBasis.ToString());
            }

            writer.WriteEndElement();
            writer.WriteStartElement("enc", "MonomerB", xmlns);
            if (encounter.EnergyCount >= 3)
            {
                writer.WriteValue(encounter.MonomerBDimerBasis.ToString());
            }

            writer.WriteEndElement();
            writer.WriteEndElement();
            writer.WriteStartElement("enc", "Basis", xmlns);
            writer.WriteStartAttribute("enc", "Type", xmlns);
            writer.WriteValue("Monomer");
            writer.WriteEndAttribute();
            writer.WriteStartElement("enc", "MonomerA", xmlns);
            if (encounter.EnergyCount >= 4)
            {
                writer.WriteValue(encounter.MonomerAMonomerBasis.ToString());
            }

            writer.WriteEndElement();
            writer.WriteStartElement("enc", "MonomerB", xmlns);
            if (encounter.EnergyCount == 5)
            {
                writer.WriteValue(encounter.MonomerBMonomerBasis.ToString());
            }

            writer.WriteEndElement();
            writer.WriteEndElement();
            writer.WriteStartElement("enc", "InteractionEnergy", xmlns);
            writer.WriteStartElement("enc", "Hartree", xmlns);
            if (encounter.EnergyCount >= 3)
            {
                writer.WriteValue(encounter.InteractionEnergyHartrees.ToString());
            }

            writer.WriteEndElement();
            writer.WriteStartElement("enc", "Kjmol", xmlns);
            if (encounter.EnergyCount >= 3)
            {
                writer.WriteValue(encounter.InteractionEnergyKjmol.ToString());
            }

            writer.WriteEndElement();
            writer.WriteEndElement();
            writer.WriteStartElement("enc", "BindingConstant", xmlns);
            if (encounter.EnergyCount >= 3)
            {
                writer.WriteValue(encounter.BindingConstant.ToString());
            }

            writer.WriteEndElement();
            writer.WriteEndElement();
            writer.Close();
            xml.Close();
        }
    }
}
